man/repart.d.xml

```diff
@@ -493,6 +493,22 @@
         specified root directory or disk image root.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>MakeDirectories=</varname></term>
+
+        <listitem><para>Takes an absolute file system path, declaring a directory to create within the new
+        file system. Behaviour is similar to <varname>CopyFiles=</varname>, but instead of copying in a set
+        of files this just creates the specified directories with the default mode of 0755 owned by the root
+        user and group, plus all their parent directories (with the same ownership and access mode). To
+        configure directories with different ownership or access mode, use <varname>CopyFiles=</varname> and
+        specify a source tree to copy containing appropriately owned/configured directories. This switch is
+        primarily useful to pre-create directory inodes that shall be over-mounted by other file system when
+        the image is used. This option may be used more than once to create multiple directories. When
+        <varname>CopyFiles=</varname> and <varname>MakeDirectories=</varname> are used together the former is
+        applied first. If a directory listed already exists no operation is executed (in particular, the
+        ownership/access mode of the directories is left as is).</para></listitem>
+      </varlistentry>
```

Comment:
Hmm, hmm, instead of building another language to support this, shouldn't tmpfiles.d be reused for this?
Because if we have MakeDirectories, maybe somebody will want to specify the mode, and the next person ownership, etc. So instead of reimplementing this, I think it might make sense to allow
```
MakeContent=d[irectory] /run/usr 0755
MakeContent=d[irectory] /root 0700
...
```

And yes, I think we should extend the tmpfiles syntax to allow longer names in the first column. It'd make the syntax here much clearer, but would also make tmpfiles more readable, and I was thinking about proposing this independently.

Reply:
So I thought about this, but I decided against this. I don't want to recreate `systemd-tmpfiles --image=` with this after all. The only reason this exists at all is so that we can put together images with `systemd-repart` that contain multiple partitions, and that we can mount them right away with `systemd-mount` or another tool. i.e. think of an image that consists of a /usr partition, where a root partition is generated at first boot. If we'd create it entirely empty and then try to mount the image (which would mean first mount the empty root partition, and the /usr partition into its /usr directory) this would fail without this option, since /usr a directory to mount over would simply not exist).

Hence, this is the absolute minimal option to make sure that an image the tool generates is instantly mountable, regardless if you use systemd-mount, systemd-dissect, systemd-nspawn or just boot it up with regularly with baremetal systemd.

I think people should use systemd-tmpfiles --image= if they want to "offline" populate images like this in more complex ways. But without MakeDirectories= there's a chicken-and-egg problem here: `systemd-tmpfiles --image=` might not be able to even mount the image to then create the files and dirs in if the mount point dirs aren't there.

I'll make this a bit clearer and in particular point point people towards using systemd-tmpfiles --image= for more complex logic.

Hope that makes sense?

Reply:
OK, makes sense.

Nitpick: the description should change "Takes an absolute file system path, declaring a directory" → "Takes one or more absolute paths, separated by whitespace, each declaring a directory" or so.